main.tf

```diff
@@ -14,11 +14,141 @@
  * limitations under the License.
  */
 
-terraform {
-  required_version = "~> 0.12.0"
+locals {
+  seed_project_id             = format("%s-%s", var.project_prefix, "seed")
+  impersonation_apis          = distinct(concat(var.activate_apis, ["serviceusage.googleapis.com", "iamcredentials.googleapis.com"]))
+  impersonation_enabled_count = var.sa_enable_impersonation == true ? 1 : 0
+  activate_apis               = var.sa_enable_impersonation == true ? local.impersonation_apis : var.activate_apis
+  org_project_creators        = distinct(concat(var.org_project_creators, ["serviceAccount:${google_service_account.org_terraform.email}", "group:${var.group_org_admins}"]))
 }
 
-resource "google_storage_bucket" "main" {
-  project = var.project_id
-  name    = var.bucket_name
+resource "random_id" "suffix" {
+  byte_length = 2
+}
+
+data "google_organization" "org" {
+  organization = var.organization_id
+}
+
+/******************************************
+  Create IaC Project
+*******************************************/
+
+module "seed_project" {
+  source                      = "terraform-google-modules/project-factory/google"
+  version                     = "~> 5.0"
+  name                        = local.seed_project_id
+  random_project_id           = "true"
+  disable_services_on_destroy = "false"
+  folder_id                   = var.folder_id
+  org_id                      = var.organization_id
+  billing_account             = var.billing_account
+  activate_apis               = local.activate_apis
+}
+
+/******************************************
+  Service Account - Terraform for Org
+*******************************************/
+
+resource "google_service_account" "org_terraform" {
+  project      = module.seed_project.project_id
+  account_id   = "org-terraform"
+  display_name = "CFT Organization Terraform Account"
+}
+
+/***********************************************
+  GCS Bucket - Terraform State
+ ***********************************************/
+
+resource "google_storage_bucket" "org_terraform_state" {
+  project       = module.seed_project.project_id
+  name          = format("%s-%s-%s", var.project_prefix, "tfstate", random_id.suffix.hex)
+  location      = var.default_region
+  force_destroy = true
```

Comment:
This seems dangerous. I think users should actively ensure that all Terraform state has been removed from the state bucket before destroying it.

Reply:
Yeah true, this was a relic of me creating / blowing up environments end to end. I'll remove this.